framework/src/reporters/ReporterData.C

```diff
@@ -15,138 +15,121 @@ ReporterData::ReporterData(MooseApp & moose_app) : _app(moose_app) {}
 void
 ReporterData::copyValuesBack()
 {
-  for (const auto & context_ptr : _context_ptrs)
-    context_ptr->copyValuesBack();
+  for (const auto & name_context_pair : _context_ptrs)
+    name_context_pair.second->copyValuesBack();
 }
 
 void
 ReporterData::finalize(const std::string & object_name)
 {
-  // FYI, for the minimum compiler 'auto' doesn't work in argument of the lambda
-  // ReporterData.C:xx:xx: error: 'auto' not allowed in lambda parameter
-  auto func = [object_name, this](const std::unique_ptr<ReporterContextBase> & ptr) {
-    if (ptr->name().getObjectName() == object_name)
-    {
-      // Perform error checking for undeclared items
-      if (this->_declare_names.find(ptr->name()) == this->_declare_names.end())
-        mooseError("The Reporter value '", ptr->name(), "' was not declared.");
-
```

Comment:
Where does this happen now? I am sad you are getting rid of my lambdas, I love these little guys.

Reply:
I think this should actually be an assertion now (didn't mean to remove it). It should be checked before hand in https://github.com/loganharbour/moose/blob/moar_moar_const/framework/src/reporters/ReporterData.C#L78